natural_language_processing/text_generation/gpt/run.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+
+from transformers import GPT2Tokenizer, GPT2Model
+
+from utils.benchmark import run_model
+from utils.misc import print_goodbye_message_and_die
+from utils.nlp.text_generation_dummy import TextGenerationDummyDataset
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Run model from Hugging Face's transformers repository"
+                                                 "for text generation task.")
+    parser.add_argument("-m", "--model_name",
+                        type=str, choices=["gpt2"], required=True,
+                        help="name of the model")
+    parser.add_argument("-b", "--batch_size",
+                        type=int, default=1,
+                        help="batch size to feed the model with")
+    parser.add_argument("-f", "--framework",
+                        type=str, choices=["pytorch"], required=True,
+                        help="specify the framework in which a model should be run")
+    parser.add_argument("--timeout",
+                        type=float, default=60.0,
+                        help="timeout in seconds")
+    parser.add_argument("--num_runs",
+                        type=int,
+                        help="number of passes through network to execute")
+    return parser.parse_args()
+
+
+def run_pytorch(model_name, batch_size, num_runs, timeout, disable_jit_freeze=False, **kwargs):
+    from utils.pytorch import PyTorchRunner
+
+    def run_single_pass(pytorch_runner, _dummy_dataset):
+        output = pytorch_runner.run(1, _dummy_dataset.get_input()['input_ids'])
```

Comment:
you set task size to 1 but what if somebody picks batch size of != 1? actually with this model it will be even more complex because we should measure task size in tokens, and this is variable across runs - this has already been solved in Daniel's llama PR so you will be able to reuse that once we merge it (https://github.com/AmpereComputingAI/ampere_model_library/pull/191)

Reply:
what I mean here is that networks like gpt generate arbitrary number of tokens per single run - sometimes they could extend prompt of "hi, ..." to "hi, how are you? <END>" and sometimes to "hi, ive been thinking about you... whatchya doinnn? blah blah blah blah ...." - therefore task_size should be equal to total number of tokens generated in given run